HIP-0006.md

```diff
@@ -0,0 +1,276 @@
+
+## Extra Notes
+
+### Bids must remain valid off-chain
+
+The partially-signed bid transactions are collected by the name owner and commit
+to spending a UTXO of some value V from the blockchain. In the simplest construction,
+nothing prevents the bidder from invalidating this off-chain transaction by
+double-spending the coin they used to bid. For this reason, the name owner must
+routinely check that all submitted bids are still valid. They can try finalizing
+the highest bid and if that fails, try to finalize the second-highest bid, etc.
+
+More advanced constructions may be possible to more effectively prevent this,
+see the next section for one  possible extension.
+
+### "Locked bids" mode
+
+To ensure that bids are funded by reliable outputs, the bidder and name owner
+can engage in a time-lock multisig contract that will ensure the bids are valid
+until the end of the auction. Bidders will effectively be "locking up" their bids
+for the duration of the auction. After the auction expiry time, bidders can "redeem" their
+losing bids with (almost) no potential grief. This pattern should be very familiar to
+Handshake users!
+
+**OPEN:** Name owners must include extra metadata in their auction announcement:
+
+1. An indication that all bids must be submitted using this mode, or they will be ignored.
+
+2. An auction expiration time, set by blockchain height: `expiration_height`
+
+3. A public key from the name owner's wallet that they can sign transactions with: `owner_pk`
+
+**BID:** When a user wants to submit a bid for this name, they follow these steps:
+
+1. Derive a public key from their wallet that they can sign with: `bidder_pk`
+
+2. Combine their key and the auction announcement data into the script below: `bid_lock_script`
+
+3. Derive the corresponding address for that script and send their bid amount to that address: `bid_lock_tx`
+
+4. Construct a partially-signed TX `bid_submit_tx` as defined in [Auction BID](#auction-bid) but with the following modifications:
+
+    1. Input 1 of this transaction will spend the UTXO created by `bid_lock_tx`
+
+    2. The redeem script in the witness of input 1 will be a partially-signed multisig spend defined below as `bid_submit_script`,
+    with a single `0x00` byte as a placeholder for the name owner's signature.
+
+5. Bidder submits the following data to the name owner:
+
+    1. `bid_lock_script` with all data inserted such that the name owner can verify it
+    matches the address in the output of `bid_lock_tx`.
+
+    2. The double-partially-signed `bid_submit_tx` (which will explicitly indicate the outpoint of the UTXO created by `bid_lock_tx`)
+
+**FINALIZE:** The name owner must finalize the atomic swap BEFORE `expiration_height`:
+
+1. Name owner chooses the highest bid and verifies that its `bid_lock_tx` is confirmed on chain,
+and that the output address is spendable by `bid_lock_script`. They construct a final TX as
+defined in [Auction FINALIZE](#auction-finalize):
```

Comment:
The name owner should also check the TX's locktime -- although @rithvikvibhu has got me thinking that more interesting constructions like "make a limited time offer" are possible with *shorter* locktimes.